specification/flag-evaluation/flag-evaluation.md

```diff
@@ -0,0 +1,195 @@
+# Flag Evaluation API
+
+## Overview
+
+The `evaluation API` allows for the retrieval of feature flag values, independent of any flag control plane or vendor. In the absence of a feature flag system implementation (a [provider](../provider/providers.md)) the `evaluation API` uses the "No-op provider", which simply returns the supplied default flag value.
+
+#### Initialization and Configuration
+
+##### Requirement 1.1
+
+> The `API`, and any state it maintains **SHOULD** exist as a global singleton, even in cases wherein multiple versions of the `API` are present at runtime.
+
+It's important that multiple instances of the `API` not be active, so that state stored therein, such as the registered `provider`, static global `evaluation context`, and globally configured `hooks` allow the `API` to behave predictably. This can be difficult in some runtimes or languages, but implementors should make their best effort that only a single instance of the `API` is used.
```

Comment:
I did not realize this was a consideration. It is not what we've seen in practice. It is generally how users work but not all of them. We have customers that instantiate multiple clients in an application.

Reply:
There's no problem with multiple clients (see 1.5). This is just a requirement for the API. A global singleton is necessary for advanced use cases, like "injecting" implementation into 3rd party libraries. If a third party implements the flag API, feature flagging can be used therein because it will be referencing the same global singleton, configured with a particular provider.